app/service.py

```diff
@@ -309,6 +317,56 @@ def post(self):
         return shopcart.serialize(), status.HTTP_201_CREATED, {'Location': location_url}
 
 
+######################################################################
+#  PATH: /shopcarts/users
+######################################################################
+@ns.route('/users')
+#@ns.param('amount', 'the min value to check the users shopcart list')
+class ShopcartUsersResource(Resource):
+    """
+    ShopcartUsersResource class
+
+    Allows getting the list of the users whose shopcarts worth more than given amount
+    GET /users - Returns the list of the users whose shopcarts' total is more than a certain amount
+    """
+    #@api.expect(amount_fields)
+    @ns.param('amount', 'amount for searching')
+
+    ############################################################################
+    # QUERY DATABASE FOR SHOPCARTS HAVING PRODUCTS WORTH MORE THAN GIVEN AMOUNT
+    ############################################################################
+    @ns.doc('get_users_with_shopcart_more_than_given_amount')
+    #@ns.marshal_list_with(string)
+    def get(self):
+        """
+        Get the shopcart list of users
+        This endpoint will show the list of the user shopcart list
+        with amount more than given amount
+        """
+        amount = request.args.get('amount');
+        app.logger.info("Request to get the list of the user shopcart having more than {}".format(amount))
+        
+        if amount is None:
+            app.logger.info("amount is none")
+            raise NotFound("Required parameter amount not found")
```

Comment:
should it be 
abort(status.HTTP_404_NOT_FOUND, 'Required parameter amount not found') ?

Reply:
got the point. done.